### PR TITLE
[ML] Fix timeout ingesting an empty string into a semantic_text field

### PR DESCRIPTION
**Pull request by davidkyle**
The sentence chunker returns 0 chunks if the input is an empty string. After chunking the Elasticsearch inference service would send an empty lists of request to the ml node for processing but the node never replied because there were no requests to process.

The first change is for the sentence chunker to return "" if the input is "", this behaviour is inline with the word based chunker. 

The second change is to protect against an empty inference request in the inference action.

The bug only applies to inference endpoints using the Elasticsearch service configured with sentence chunking. Sentence chunking which was introduced in 8.16

### CODE REVIEW

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunker.java [hunk 1]
```diff
@@ -66,6 +66,10 @@ public List<String> chunk(String input, ChunkingSettings chunkingSettings) {
      * @return The input text chunked
      */
     public List<String> chunk(String input, int maxNumberWordsPerChunk, boolean includePrecedingSentence) {
+        if (input.isEmpty()) {
+            return List.of("");
+        }
```
Comment (Mikep86) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunker.java:71]: Do we need to trim the input before checking if it's empty? How does the chunker handle input that is only whitespace?

Comment (davidkyle) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/chunking/SentenceBoundaryChunker.java:71]: Good catch @Mikep86 thanks. 

There's a whole class of bugs for "things that don't chunk". Pure whitespace, a single character or the same character repeated thousands of times don't chunk. I've added test cases for these situations and the solution I've implemented here in this is to return the original input if it did not chunk.  This applies to both the Word and Sentence chunker.

This makes me wonder if there should be an upper limit on the chunk size in terms of number of characters. A badly formed input contain latin characters but no whitespace would result in a single large chunk, think a binary file base64 encoded.

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunkerTests.java [hunk 2]
```diff
@@ -31,16 +32,62 @@
...
 
-    public void testBlankInput() {
+    public void testEmptyInput_SentenceChunker() {
+        var embeddingType = randomFrom(EmbeddingRequestChunker.EmbeddingType.values());
+        var batches = new EmbeddingRequestChunker(List.of(), 10, embeddingType, new SentenceBoundaryChunkingSettings(250, 1))
+            .batchRequestsWithListeners(testListener());
+        assertThat(batches, empty());
+    }
+
+    public void testWhitespaceInput_SentenceChunker() {
```
Comment (Mikep86) [inline x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunkerTests.java:48]: Do we need a test for whitespace input for the word chunker?

**Review by Mikep86 — APPROVED** [review]
Looks good, thanks for handling those edge cases

File: x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunkerTests.java (hunk 2 above)
Comment (davidkyle) [reply x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/chunking/EmbeddingRequestChunkerTests.java:48]: Covered by an existing test

**Pull request merged**